apps/condo/domains/user/schema/ConfirmPhone.js

```diff
@@ -0,0 +1,385 @@
+const { v4: uuid } = require('uuid')
+const faker = require('faker')
+const access = require('@core/keystone/access')
+const conf = require('@core/config')
+const { Text, DateTimeUtc, Checkbox, Integer } = require('@keystonejs/fields')
+const { historical, uuided, softDeleted } = require('@core/keystone/plugins')
+const { GQLListSchema, GQLCustomSchema } = require('@core/keystone/schema')
+const { sendMessage } = require('@condo/domains/notification/utils/serverSchema')
+const { normalizePhone } = require('@condo/domains/common/utils/phone')
+const { SENDER_FIELD, DV_FIELD } = require('@condo/domains/common/schema/fields')
+const { has, isEmpty } = require('lodash')
+
+const { 
+    CONFIRM_PHONE_ACTION_EXPIRED,
+    CONFIRM_PHONE_SMS_CODE_EXPIRED,
+    CONFIRM_PHONE_SMS_CODE_VERIFICATION_FAILED, 
+    CONFIRM_PHONE_SMS_CODE_MAX_RETRIES_REACHED, 
+} = require('@condo/domains/user/constants/errors')
+
+const { COUNTRIES, RUSSIA_COUNTRY } = require('@condo/domains/common/constants/countries')
+const { SMS_VERIFY_CODE_MESSAGE_TYPE } = require('@condo/domains/notification/constants')
+
+const { 
+    SMS_CODE_LENGTH, 
+    SMS_CODE_TTL, 
+    CONFIRM_PHONE_ACTION_EXPIRY, 
+    CONFIRM_PHONE_SMS_MAX_RETRIES,
+} = require('@condo/domains/user/constants/common')
+
+const whiteList = conf.SMS_WHITE_LIST ? JSON.parse(conf.SMS_WHITE_LIST) : {}
```

Comment:
How can we change whiteList at runtime? Guess we need to think about bot integration.

Reply:
This white list is for developers only. When you add a phone with a code to the .env file - no SMS will be sent via the SMS gate
It works similarly to a firebase whitelist config